src/twisted/application/internet.py

```diff
@@ -197,34 +197,82 @@ def _getConnection(self):
 
 
 
-_doc={
-'Client':
-"""Connect to %(tran)s
+_doc = {
+    'Client':
+    """Connect to {tran}
 
-Call reactor.connect%(tran)s when the service starts, with the
+Call reactor.connect{tran} when the service starts, with the
 arguments given to the constructor.
 """,
-'Server':
-"""Serve %(tran)s clients
+    'Server':
+    """Serve {tran} clients
 
-Call reactor.listen%(tran)s when the service starts, with the
+Call reactor.listen{tran} when the service starts, with the
 arguments given to the constructor. When the service stops,
 stop listening. See twisted.internet.interfaces for documentation
 on arguments to the reactor method.
 """,
 }
 
-for tran in 'TCP UNIX SSL UDP UNIXDatagram Multicast'.split():
-    for side in 'Server Client'.split():
-        if tran == "Multicast" and side == "Client":
-            continue
-        if tran == "UDP" and side == "Client":
-            continue
-        base = globals()['_Abstract'+side]
-        doc = _doc[side] % vars()
 
-        klass = type(tran+side, (base,), {'method': tran, '__doc__': doc})
-        globals()[tran+side] = klass
+
+class TCPServer(_AbstractServer):
+    __doc__ = _doc['Server'].format(tran='TCP')
```

Comment:
While this is a lot simpler than before, this expression is still too complex for pydoctor to be able to compute the docstring using static analysis.

If you could split `_doc` into `_serverDoc` and `_clientDoc`, that would simplify the expression. I checked and found no dynamic lookups in `_doc` in the module.

Reply:
Fixed